src/ru/hotmanga/src/eu/kanade/tachiyomi/extension/ru/hotmanga/HotManga.kt

```diff
@@ -0,0 +1,327 @@
+        val apiString = "$apiPathVal/catalog?orderBy=-id&page=$pageF"
+        val url = "${baseUrl}$apiString".toHttpUrl().newBuilder()
+        return GET(url.build(), headers)
+    }
+
+    override fun mangaDetailsParse(response: Response): SManga = throw NotImplementedError("Unused")
+
+    override fun pageListParse(response: Response): List<Page> = throw NotImplementedError("Unused")
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        if (response.code != 200) {
+            return MangasPage(emptyList(), false)
+        }
+
+        val values = json.parseToJsonElement(response.body.string()).jsonArray
+        val mangaListDto = mutableListOf<MangaDto>()
+        for (item in values) {
+            try {
+                mangaListDto.add(json.decodeFromJsonElement<MangaDto>(item))
+            } catch (e: Exception) {
+                Log.i("HotManga", e.toString())
+            }
+        }
+        var hasNextPage = true
+        if (mangaListDto.isEmpty()) {
+            hasNextPage = false
+        }
+        val mangas = mutableListOf<SManga>()
+        for (mangaItem in mangaListDto) {
+            val element = mangaItem.toSManga()
+            mangas.add(element)
+        }
+        return MangasPage(mangas, hasNextPage)
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        // TODO Add filters, use page param, investigate limit param
+        val apiPathVal = apiPathsMap[baseUrl]
+        val apiString = "$apiPathVal/books/search?filter[query]=$query&limit=24"
+        val url = "${baseUrl}$apiString".toHttpUrl().newBuilder()
+        return GET(url.build(), headers)
+    }
+
+    override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> {
+        val chapters = mutableListOf<SChapter>()
+        var page = 0
+        val mangaUrl = manga.url
+        val bookId = URI(mangaUrl).findParameterValue("bookId")
+        val countChapters = URI(mangaUrl).findParameterValue("countChapters")?.toLong()
+        val apiPathVal = apiPathsMap[baseUrl]
+        while (chapters.size < countChapters!!) {
+            val urlBase =
+                "$baseUrl$apiPathVal/chapters/with-branches?filter%5BbookId%5D=$bookId&page=$page"
+            val request = GET(urlBase.toHttpUrl(), headers)
+            val body = client.newCall(request).execute().body.string()
+            val values = json.parseToJsonElement(body).jsonArray
+
+            if (values.isEmpty()) {
+                break
+            }
+
+            for (item in values) {
+                val number = item.jsonObject["number"].toString().replace("\"", "").toFloat()
+                val createdAt = item.jsonObject["createdAt"]?.jsonPrimitive?.content
+                val tom = item.jsonObject["tom"]?.jsonPrimitive?.content
+                val id = item.jsonObject["id"].toString()
+                val chapterBranches = item.jsonObject["chapterBranches"]?.jsonArray
```

Comment:
Use a DTO instead of this. You can use this extension for help with that: https://plugins.jetbrains.com/plugin/7834-dto-generator